### DISCUSSION TITLE
Best practice for using Typescript with GraphQL (generated files)

### DISCUSSION THREAD

**Discussion by charklewis** [dotansimha/graphql-code-generator#6392] [Q&A]
I am new to building a dockerised set up and not too sure how to solve a dependancy issue. 

I have two repos, one to manage my database and the other to manage my action handlers (lambda functions as an express app). The action handlers use the database to generate a graphql scheme, and this is used to type check the code when the action handlers express app starts (as it is written in Typescript). 

However, I have heard (and I am finding it really hard to implement) to have one docker container depending on another. A suggestion I have had from a friend is to save the generated graphql schema to GitHub, however the schema is tens of thousands of lines long and it really should depend on the database being composed in the docker image. 

Has anyone had this issue before or any tips on how to resolve?

**Comment by dotansimha** [comment]
Hi! 
There are many ways of loading your schema. It can be directly from a local file, from GitHub, a running server (using URL), and more. Check the docs for more information: https://www.graphql-code-generator.com/docs/getting-started/schema-field 

I usually prefer monorepo setup, and then my code always has a local file(s) with the schema, that I can just load locally.
